crates/gitbutler-tauri/src/open.rs

```diff
@@ -0,0 +1,75 @@
+use crate::error::Error;
+use anyhow::{bail, Context};
+use std::env;
+use tracing::instrument;
+use url::Url;
+
+pub(crate) fn open_that(path: &str) -> anyhow::Result<()> {
+    let target_url = Url::parse(path).with_context(|| format!("Invalid path format: '{path}'"))?;
+    if !["http", "https", "mailto", "vscode", "vscodium"].contains(&target_url.scheme()) {
+        bail!("Invalid path scheme: {}", target_url.scheme());
+    }
+
+    fn clean_env_vars<'a, 'b>(
+        var_names: &'a [&'b str],
+    ) -> impl Iterator<Item = (&'b str, String)> + 'a {
+        var_names
+            .iter()
+            .filter_map(|name| env::var(name).map(|value| (*name, value)).ok())
+            .map(|(name, value)| {
+                (
+                    name,
+                    value
+                        .split(':')
+                        .filter(|path| {
+                            !path.contains("appimage-run") && !path.contains("/tmp/.mount")
+                        })
+                        .collect::<Vec<_>>()
+                        .join(":"),
+                )
+            })
+    }
+
+    let mut cmd_errors = Vec::new();
+
+    for mut cmd in open::commands(path) {
+        let cleaned_vars = clean_env_vars(&[
+            "APPDIR",
+            "GDK_PIXBUF_MODULE_FILE",
+            "GIO_EXTRA_MODULES",
+            "GIO_EXTRA_MODULES",
+            "GSETTINGS_SCHEMA_DIR",
+            "GST_PLUGIN_SYSTEM_PATH",
+            "GST_PLUGIN_SYSTEM_PATH_1_0",
+            "GTK_DATA_PREFIX",
+            "GTK_EXE_PREFIX",
+            "GTK_IM_MODULE_FILE",
+            "GTK_PATH",
+            "LD_LIBRARY_PATH",
+            "PATH",
+            "PERLLIB",
+            "PYTHONHOME",
+            "PYTHONPATH",
+            "QT_PLUGIN_PATH",
+            "XDG_DATA_DIRS",
+        ]);
+
+        cmd.envs(cleaned_vars);
+        cmd.current_dir(env::temp_dir());
+        if cmd.status().is_ok() {
+            return Ok(());
+        } else {
+            cmd_errors.push(anyhow::anyhow!("Failed to execute command {:?}", cmd));
```

Comment:
Good call, I think that will help to reproduce issues on systems that fail for all possible commands.